kotest-property/src/commonMain/kotlin/io/kotest/property/checkall.kt

```diff
@@ -0,0 +1,32 @@
+package io.kotest.property
+
+// todo have these methods delegate to one that uses PropTestConfig
+fun checkAll(iterations: Int, seed: Long, test: PropertyContext.() -> Unit): PropertyContext {
+   val context = PropertyContext(RandomSource.seeded(seed))
+   repeat(iterations) {
+      try {
+         context.test()
+      } catch (t: Throwable) {
+         // shrink here then throw
```

Comment:
@myuwono shrinking here

Reply:
this is actually harder than i originally anticipated.. ive been thinking about this but it's so hard to make it reliable for inline created arb.. i'll share some of my findings soon.

in a nutshell if the test has something along these lines, it's somewhat difficult to replay the lambda block with the injected shrunk value unless we can have a reliable mapping between `Arb<*> -> RTree<*>`. We can't do that right now as hashCode of an Arb<*> will always be different all the time.. Neither can we infer that from the value type (see else block on how c was generated)
```kotlin
checkAll {
  val a = Arb.string(...).value() // new arb
  val b = anotherArb(a).value() // dependent arb
  if (somePredicate(b)) {
    val c: C = yetAnotherArb(a, b).value() // another arb inside an if, 
    val expected = ...
    c shouldBe ... // consider the test fails here
  } else {
    val c: C = notTheSameCArbButTheSameType(a, b).value() // another arb that is not the same third arb
    val d: D = arbD(a, c).value() // arbD, a fourth arb that only exists in the else block
    val expected = ...
    d shouldBe ...
  }
}
```